File: client-clojure/src/sirqul_io_t_platform/specs/shipment_import_status.clj
(ns sirqul-io-t-platform.specs.shipment-import-status
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.shipment :refer :all]
            )
  (:import (java.io File)))


(def shipment-import-status-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :name) string?
   (ds/opt :batchId) string?
   (ds/opt :message) string?
   (ds/opt :shipment) shipment-spec
   (ds/opt :started) inst?
   (ds/opt :completed) inst?
   })

(def shipment-import-status-spec
  (ds/spec
    {:name ::shipment-import-status
     :spec shipment-import-status-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/ShipmentImportStatus.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import java.time.OffsetDateTime

import ShipmentImportStatus._

case class ShipmentImportStatus (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
name: Option[String],
batchId: Option[String],
message: Option[String],
shipment: Option[Shipment],
started: Option[OffsetDateTime],
completed: Option[OffsetDateTime])

object ShipmentImportStatus {
  import DateTimeCodecs._

  implicit val ShipmentImportStatusCodecJson: CodecJson[ShipmentImportStatus] = CodecJson.derive[ShipmentImportStatus]
  implicit val ShipmentImportStatusDecoder: EntityDecoder[ShipmentImportStatus] = jsonOf[ShipmentImportStatus]
  implicit val ShipmentImportStatusEncoder: EntityEncoder[ShipmentImportStatus] = jsonEncoderOf[ShipmentImportStatus]
}
